hub/core/chunk_engine/write.py

```diff
@@ -89,7 +99,12 @@ def write_bytes(
             `IndexMeta.add_entry` supports more parameters than this. Anything passed in this dict will also be used
             to call `IndexMeta.add_entry`.
     """
-
+    compression = extra_sample_meta.get("compression", False)
+    if not compression and tensor_meta.compression is not None:
+        compression = tensor_meta.compression
+        extra_sample_meta["compression"] = compression
+        compressor = get_compressor(compression)
+        b = _compress_bytes(b, compressor)
```

Comment:
i know i suggested this be a separate function, but `_compress_bytes` doesn't give me much information, could you just replace this with:

```python
if compressor:
    b = compressor.encode(b)
```